cargo/private/runfiles_enabled.bzl

```diff
@@ -0,0 +1,167 @@
+"""A small utility module dedicated to detecting whether or not the `--enable_runfiles` and `--windows_enable_symlinks` flag are enabled
+"""
+
+load("@bazel_skylib//lib:selects.bzl", "selects")
+load("@bazel_skylib//rules:common_settings.bzl", "bool_setting")
+
+RunfilesEnabledInfo = provider(
+    doc = "A singleton provider that contains the raw value of a build setting",
+    fields = {
+        "value": "The value of the build setting in the current configuration. " +
+                 "This value may come from the command line or an upstream transition, " +
+                 "or else it will be the build setting's default.",
+    },
+)
+
+def _runfiles_enabled_setting_impl(ctx):
+    return RunfilesEnabledInfo(value = ctx.attr.value)
+
+runfiles_enabled_setting = rule(
+    implementation = _runfiles_enabled_setting_impl,
+    doc = "A bool-typed build setting that cannot be set on the command line",
+    attrs = {
+        "value": attr.bool(
+            doc = "A boolean value",
+            mandatory = True,
+        ),
+    },
+)
+
+_RUNFILES_ENABLED_ATTR_NAME = "_runfiles_enabled"
+
+def runfiles_enabled_attr(default = Label("//cargo/private:runfiles_enabled")):
+    return {
+        _RUNFILES_ENABLED_ATTR_NAME: attr.label(
+            doc = "A flag representing whether or not runfiles are enabled.",
+            providers = [RunfilesEnabledInfo],
+            default = default,
+            cfg = "exec",
+        ),
+    }
+
+def runfiles_enabled_build_setting(name, **kwargs):
```

Comment:
Nice hack ;-)

I think that this can be simplified since you don't need a single build setting to match both the platform and the value of `--enable_runfiles`. This currently also matches the target platform, but the default is based on the host platform.

How about only matching the flag here and then using `ctx.configuration.host_path_separator` to filter out a Windows host in `runfiles_enabled_build_setting`?

Reply:
This took an embarrassing amount of brain power 😅 , could I get you to slice and dice this code to be the simpler form?